api/models.py

```
@@ -409,22 +404,191 @@ class User(ModelOpsMixin):
+        if not self.id:
+            self.save()
+
+        return dict(id=self.id,
+                    message=self.message,
+                    createdAt=str(self.created_at),
+                    category=self.category,
+                    read=bool(self.read),
+                    recipientId=self.recipient_id,
+                    sender={"userId": self.user_id,
+                            "userName": self.user_name,
+                            "userImg": self.user_img},
+                    link=self.resource_id)
+
+    @classmethod
+    def create(cls, message, category, resource_id, recipient_id, **sender):
+        """Factory method for initiliazing notifications.
+
+        Args:
+          message(str): notification message
+          category(str): notification's categorty e.g contribution
+          resource-id(str): specify source of notification
+          recipient_id: recepient to be notified
+          *sender(dict): sender's information
+
+        Returns:
+          Notification()
+        """
+        return cls(
+                message=message,
+                category=category,
+                resource_id=resource_id,
+                recipient_id=recipient_id,
+                user_id=sender.get('userId'),
+                user_name=sender.get('userName'),
+                user_img=sender.get('userImg'),
+                created_at=date.today())
+
+    @classmethod
+    def toggle_read(cls, recipient_id, resource_id=None):
+        """Toggle read status of notification.
+
+        Args:
+          recipient_id(str): recipient's id
+          resource_id(str): specify id for resource to be updated
+
+        Returns:
+          notification(dict)
+          Raises KeyError for resource_id not found
+        """
+        if resource_id:
+            notification = cls.load(resource_id)
+            if notification.recipient_id != recipient_id:
+                return False
```

What happens if `recipient_id` is not provided?

---

@wanjiku-carol, I believe `recipient_id` is required as against `resource_id` which can be None. This should throw an error. Am I right, @Mnickii?

---

yes, you are right @andela-milesanmi. The recipient_id is necessary to update all the notifications of a particular recipient to read, else one can provide the particular notification_id to update status to read. It is a positional argument hence if not provided, it will throw a type error. @wanjiku-carol